src/gate_detect/test/gate-detection-test.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: May 5th, 2018
+ * Description: Test functionality of GateDetection
+ */
+
+#include "GateDetection.h"
+#include "testUtilities/TestUtils.h"
+#include <gtest/gtest.h>
+
+using namespace cv;
+using namespace std;
+
+TEST(TestSuite, testCase1) {
+    Mat image;
+
+    image = imread("testImages/gateFrontOrange.png",
+                   CV_LOAD_IMAGE_COLOR); // Read the file
+
+    if (image.empty()) // Check for invalid input
+    {
+        cout << "Could not open or find the image" << std::endl;
+        FAIL();
+    }
```

Comment:
We seem to get an image and check if it exists a lot. Might be helpful to add a function that does this to the top of this file to avoid repetitive code.